### DISCUSSION TITLE
How to validate a string lenth scope in cue

### DISCUSSION THREAD

**Discussion by chenlinli** [cue-lang/cue#2051] [Q&A] [answered]
we need to check a json schema below,how to check the "name" field's lenth scope between 2 and 10 by cue
{
	"aa": 1,
	"fruits": [{
		"name": "a",
        "age":13
	}, {
		"name": "bgggg",
      	"add":"fgfgfjjjkj"
	}]
}

**Comment by slewiskelly** [comment, marked as answer]
You can add constraints using the `strings` package:

```cue
import "strings"

name: strings.MinRunes(2) & strings.MaxRunes(10)
```

Or you could use a regular expression:

```cue
name: string & =~ #"\w{2,10}"#
```

**Reply by chenlinli** [reply]
It's seems doesn't work in golang.
First I write a cue File
awene.cue
```
import ("strings")

#schema:{
	aweme_list:[...#AwemeStrcut]
}

#AwemeStrcut:{
	desc:string & strings.MinRunes(2) & strings.MaxRunes(10)
}
```
and I want to use the cue shema to validate a json from File in golang,below is my go test case
```
func TestStrLen(t *testing.T) {
	val, _ := os.ReadFile("resource/aweme.json")

	s, _ := os.ReadFile("resource/aweme.cue")
	ctx := cuecontext.New()
	s2 := ctx.CompileBytes(s, cue.Filename("schema2.cue"))
	v := ctx.CompileBytes(val, cue.Scope(s2),cue.Filename("val.cue"))
	u := s2.Unify(v)
	if  u.Err()!=nil{
		msg := errors.Details(v.Err(), nil)
		fmt.Printf("Unify Error:\n%s2\n", msg)
	}
	err := u.Validate()
	if err != nil {
		msg := errors.Details(err, nil)
		fmt.Printf("Validate Error:\n%s2\n", msg)
	}
	fmt.Printf("%#v\n", u)

}
```
and aweme.json is :
```
{
  "status_code": 0,
  "min_cursor": "0",
  "max_cursor": "0",
  "has_more": 1,
  "aweme_list": [
    {
      "aweme_id": "7153041669548494114",
      "desc": "猫：这辈子没这么无语过#小猫99999999999咪带娃日记",
      "create_time": "1665447307",
      }]
}
```

obviously,the length of "desc" is greater than 10, but the result has no error

**Reply by myitcv** [reply]
@chenlinli for multiline code blocks please use something like this:

    ```
    here is some code
    ```

As you can see, your comments otherwise do not format properly and it's very hard to read. 

Thanks

**Reply by chenlinli** [reply]
sorry,I am new to the gitlab discussion,my reply is fixed now

**Comment by chenlinli** [comment]
这是来自QQ邮箱的假期自动回复
谢谢，您的邮件我已收到。

**Comment by praveentiru** [comment]
@slewiskelly I am new to CUE. I am bit confused. I checked standard golang strings package and there are no functions by name MinRunes and MaxRunes. What strings package are we referring here?

**Reply by praveentiru** [reply]
After a little more digging, I have found the answer. I believe you are referring to packages in pkg which refers to packages that are accessible from CUE. Refer pkg here -> [Integration with Go](https://cuelang.org/docs/integrations/go/)
